### PR TITLE
UI40: Remove StatusBarContext and HOC

### PR DESCRIPTION
**Pull request by raplemie**
This finalize iTwin/itwinjs-backlog#190 by removing the dead code from the packages.

**This is merging to branch appui-4.0, NOT master**

### CODE REVIEW

File: docs/changehistory/NextVersion.md [hunk 1]
```diff
@@ -29,3 +29,18 @@ The synchronous `void`-returning overload of [IModelTransformer.initFromExternal
 It will still perform the old behavior synchronously until it is removed. It will now however return a `Promise` (which should be
 awaited) if invoked with the an [InitFromExternalSourceAspectsArgs]($transformer) argument, which is necessary when processing
 changes instead of the full source contents.
+
+## AppUI
```
Comment (aruniverse) [inline docs/changehistory/NextVersion.md:33]: i would make a separate appui 4.0 next ver md for now

**Comment by raplemie** [comment]
@NancyMcCallB The full build was skipped, is that what is expected ? Seems to be risky if we dont run anything on this branch.

File: docs/changehistory/NextVersion.md (hunk 1 above)
Comment (raplemie) [reply docs/changehistory/NextVersion.md:33]: This is mostly a placeholder until we figure how this is going to be, I dont mind creating a different file if that's better.

**Review by aruniverse — COMMENTED** [review]
thoughts on changing the codeowners for this appui-4.0 branch to ignore requestion non appui team reviewers until its ready to merge?

shoulkd the first pr into this branch eb setting up the 4.0 branch with a separate version policy for these pkgs?

**Comment by raplemie** [comment]
Reading through https://github.com/iTwin/itwinjs-backlog/issues/283, this will not be the first PR to go in 4.0, there is still maintenance stuff to be done before we start merging, I'll put this one back to Draft in the mean time.

**Comment by raplemie** [comment]
Moved to iTwin/appui#3

**Pull request closed without merging**